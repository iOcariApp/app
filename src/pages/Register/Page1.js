import React from "react";
import PropTypes from "prop-types";
import styles from "./Register.style";

import TextInput from "components/TextInput";
import Button from "components/Button";
import AutoFocusForm from "components/AutoFocusForm";

import availableNickname from "utils/validation/availableNickname";
import validEmail from "utils/validation/validEmail";
import passwordStrength from "utils/validation/passwordStrength";

const Page1 = ({
  values: { nickname, email, password, passwordDouble },
  onChangeValue,
  samePassword,
  goNext,
}) => (
  <AutoFocusForm style={styles.container}>
    <TextInput
      keyLabel="nickname"
      label="Nombre de jugador"
      value={nickname}
      onChangeValue={onChangeValue}
      validation={availableNickname}
    />
    <TextInput
      keyLabel="email"
      label="E-mail"
      value={email}
      onChangeValue={onChangeValue}
      validation={validEmail}
      keyboardType="email-address"
    />
    <TextInput
      keyLabel="password"
      label="Contraseña"
      value={password}
      onChangeValue={onChangeValue}
      validation={passwordStrength}
      secureTextEntry
    />
    <TextInput
      keyLabel="passwordDouble"
      label="Repetir contraseña"
      value={passwordDouble}
      onChangeValue={onChangeValue}
      validation={samePassword}
      secureTextEntry
    />
    <Button title="SIGUIENTE" style={styles.marginButtons} onPress={goNext} />
  </AutoFocusForm>
);

Page1.propTypes = {
  values: PropTypes.shape({
    nickname: PropTypes.string,
    email: PropTypes.string,
    password: PropTypes.string,
    passwordDouble: PropTypes.string,
  }),
  onChangeValue: PropTypes.func,
  samePassword: PropTypes.func,
  goNext: PropTypes.func,
};

export default Page1;
