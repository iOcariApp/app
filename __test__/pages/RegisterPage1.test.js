import React from "react";

import Page1 from "pages/Register/Page1";

const props = {
  values: {
    nickname: "",
    email: "",
    password: "",
    passwordDouble: "",
  },
  onChangeValue: noop,
  samePassword: noop,
  goNext: noop,
};

test("renders correctly", () => {
  testSnapshot(<Page1 {...props} />);
});
