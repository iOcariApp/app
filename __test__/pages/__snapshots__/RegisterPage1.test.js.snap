// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "marginBottom": 17,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "column",
          "justifyContent": "center",
          "position": "relative",
          "width": "100%",
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        data-test="input-main"
        hitSlop={undefined}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignContent": "center",
            "backgroundColor": "white",
            "borderBottomWidth": 1,
            "borderColor": "#7C7C7C",
            "borderTopLeftRadius": 4,
            "borderTopRightRadius": 4,
            "flexDirection": "row",
            "height": 56,
            "paddingBottom": 10,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 10,
            "width": "100%",
          }
        }
        testID={undefined}
      >
        <View
          style={
            Object {
              "flex": 1,
              "position": "relative",
              "top": -7,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            collapsable={undefined}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#7C7C7C",
                "fontSize": 16,
                "position": "relative",
                "top": 14,
              }
            }
          >
            Nombre de jugador
          </Text>
          <TextInput
            allowFontScaling={true}
            keyLabel="nickname"
            onBlur={[Function]}
            onChangeText={[Function]}
            onChangeValue={[MockFunction]}
            onFocus={[Function]}
            style={
              Object {
                "fontSize": 16,
              }
            }
            underlineColorAndroid="transparent"
            validation={[Function]}
            value=""
          />
        </View>
      </View>
    </View>
  </View>
  <View
    onLayout={[Function]}
    style={
      Object {
        "marginBottom": 17,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "column",
          "justifyContent": "center",
          "position": "relative",
          "width": "100%",
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        data-test="input-main"
        hitSlop={undefined}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignContent": "center",
            "backgroundColor": "white",
            "borderBottomWidth": 1,
            "borderColor": "#7C7C7C",
            "borderTopLeftRadius": 4,
            "borderTopRightRadius": 4,
            "flexDirection": "row",
            "height": 56,
            "paddingBottom": 10,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 10,
            "width": "100%",
          }
        }
        testID={undefined}
      >
        <View
          style={
            Object {
              "flex": 1,
              "position": "relative",
              "top": -7,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            collapsable={undefined}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#7C7C7C",
                "fontSize": 16,
                "position": "relative",
                "top": 14,
              }
            }
          >
            E-mail
          </Text>
          <TextInput
            allowFontScaling={true}
            keyLabel="email"
            keyboardType="email-address"
            onBlur={[Function]}
            onChangeText={[Function]}
            onChangeValue={[MockFunction]}
            onFocus={[Function]}
            style={
              Object {
                "fontSize": 16,
              }
            }
            underlineColorAndroid="transparent"
            validation={[Function]}
            value=""
          />
        </View>
      </View>
    </View>
  </View>
  <View
    onLayout={[Function]}
    style={
      Object {
        "marginBottom": 17,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "column",
          "justifyContent": "center",
          "position": "relative",
          "width": "100%",
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        data-test="input-main"
        hitSlop={undefined}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignContent": "center",
            "backgroundColor": "white",
            "borderBottomWidth": 1,
            "borderColor": "#7C7C7C",
            "borderTopLeftRadius": 4,
            "borderTopRightRadius": 4,
            "flexDirection": "row",
            "height": 56,
            "paddingBottom": 10,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 10,
            "width": "100%",
          }
        }
        testID={undefined}
      >
        <View
          style={
            Object {
              "flex": 1,
              "position": "relative",
              "top": -7,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            collapsable={undefined}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#7C7C7C",
                "fontSize": 16,
                "position": "relative",
                "top": 14,
              }
            }
          >
            Contraseña
          </Text>
          <TextInput
            allowFontScaling={true}
            keyLabel="password"
            onBlur={[Function]}
            onChangeText={[Function]}
            onChangeValue={[MockFunction]}
            onFocus={[Function]}
            secureTextEntry={true}
            style={
              Object {
                "fontSize": 16,
              }
            }
            underlineColorAndroid="transparent"
            validation={[Function]}
            value=""
          />
        </View>
      </View>
    </View>
  </View>
  <View
    onLayout={[Function]}
    style={
      Object {
        "marginBottom": 17,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "column",
          "justifyContent": "center",
          "position": "relative",
          "width": "100%",
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        data-test="input-main"
        hitSlop={undefined}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignContent": "center",
            "backgroundColor": "white",
            "borderBottomWidth": 1,
            "borderColor": "#7C7C7C",
            "borderTopLeftRadius": 4,
            "borderTopRightRadius": 4,
            "flexDirection": "row",
            "height": 56,
            "paddingBottom": 10,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 10,
            "width": "100%",
          }
        }
        testID={undefined}
      >
        <View
          style={
            Object {
              "flex": 1,
              "position": "relative",
              "top": -7,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            collapsable={undefined}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#7C7C7C",
                "fontSize": 16,
                "position": "relative",
                "top": 14,
              }
            }
          >
            Repetir contraseña
          </Text>
          <TextInput
            allowFontScaling={true}
            keyLabel="passwordDouble"
            onBlur={[Function]}
            onChangeText={[Function]}
            onChangeValue={[MockFunction]}
            onFocus={[Function]}
            secureTextEntry={true}
            style={
              Object {
                "fontSize": 16,
              }
            }
            underlineColorAndroid="transparent"
            validation={[MockFunction]}
            value=""
          />
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "marginLeft": 15,
          "marginRight": 15,
        },
        undefined,
        Array [
          Object {
            "marginBottom": 0,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 0,
            "width": "100%",
          },
          Object {
            "marginBottom": 20,
            "marginTop": 20,
          },
        ],
        Object {
          "borderRadius": 4,
        },
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        pointerEvents="box-only"
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#9E9E9E",
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 19,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Object {
              "backgroundColor": "#F50057",
            },
            Object {
              "borderRadius": 4,
            },
            Object {
              "padding": 12,
            },
            undefined,
            undefined,
            undefined,
            Array [
              Object {
                "height": 56,
              },
              Object {},
            ],
            undefined,
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {},
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "white",
                  "fontSize": 20,
                },
                undefined,
                Object {
                  "fontSize": 17.5,
                },
                Object {
                  "fontSize": 18,
                },
                undefined,
                undefined,
                undefined,
                undefined,
              ],
            ]
          }
        >
          SIGUIENTE
        </Text>
      </View>
    </View>
  </View>
</View>
`;
